qiskit/algorithms/gradients/lin_comb_qfi.py

```diff
@@ -0,0 +1,197 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+A class for the Quantum Fisher Information.
+"""
+
+from __future__ import annotations
+
+from typing import Sequence
+
+import numpy as np
+
+from qiskit.algorithms import AlgorithmError
+from qiskit.circuit import Parameter, ParameterExpression, QuantumCircuit
+from qiskit.opflow import PauliSumOp
+from qiskit.primitives import BaseEstimator
+from qiskit.primitives.utils import init_observable
+from qiskit.quantum_info import Pauli, SparsePauliOp
+from qiskit.quantum_info.operators.base_operator import BaseOperator
+
+from .base_qfi import BaseQFI
+from .qfi_result import QFIResult
+from .estimator_gradient_result import EstimatorGradientResult
+from .utils import  _make_lin_comb_qfi_circuit
+from .lin_comb_estimator_gradient import LinCombEstimatorGradient
+
+Pauli_Z = Pauli("Z")
+
+
+class LinCombQFI(BaseQFI):
+    def __init__(
+        self,
+        estimator: BaseEstimator,
+        phase_fix: bool = True,
+        derivative: str = "real",
+        **run_options,
+    ):
+        r"""Computes the Quantum Fisher Information (QFI) given a pure,
+        parameterized quantum state, where QFI is:
+
+        .. math::
+
+            \mathrm{QFI}_{kl}= 4 \mathrm{Re}[\langle \partial_k \psi | \partial_l \psi \rangle
+                - \langle\partial_k \psi | \psi \rangle \langle\psi | \partial_l \psi \rangle].
+
+        Args:
+            estimator: The estimator used to compute the gradients.
+            phase_fix: Whether to calculate the second term (phase fix) of the QFI. Default to True.
+            derivative: The type of derivative. Can be either "real", "imag", or "complex".
+                Defaults to "real".
+            run_options: Backend runtime options used for circuit execution. The order of priority is:
+                run_options in ``run`` method > gradient's default run_options > primitive's default
+                setting. Higher priority setting overrides lower priority setting.
+        """
+        super().__init__(estimator, **run_options)
+        self._phase_fix = phase_fix
+        self._derivative = derivative
+        self._gradient = LinCombEstimatorGradient(estimator, derivative="complex", **run_options)
```

Comment:
It would be helpful to see more explanation how `derivative` argument is different from `derivative` keyword in `LinCombEstimatorGradient`.